src/charm.py

```diff
@@ -287,10 +287,9 @@ def _get_istioctl(self) -> Istioctl:
         # (see https://istio.io/latest/docs/tasks/observability/logs/access-log/)
         setting_overrides["meshConfig.accessLogFile"] = "/dev/stdout"
 
-        # Configure CNI
-        # (see https://istio.io/latest/docs/setup/additional-setup/cni/#additional-configuration)
-        setting_overrides["values.cni.cniBinDir"] = self.parsed_config["cni-bin-dir"]
```

Comment:
I'm not sure if this was on purpose, but I think the configs are still exposed in cni-bin-dir, yet they now do nothing.

Also, afaik istio [didn't drop](https://istio.io/latest/docs/ambient/install/platform-prerequisites/#k3s) these configs and they seem kinda useful for custom, non-bundled CNI, so i'm kinda proposing exposing both options to the user with cni configs defaulting to `None` and only applied when configured

Reply:
Good catch, I forgot to actually delete these from the config section of charmcraft.yaml.  Not ignoring your objection, but just to make the current code consistent with itself I've pushed a change that removes these from the charmcraft.yaml as was intended

Reply:
Agreed that `cni-*-dir` configs can be useful.  I asked myself the same question of "should we just keep these?"

I'm a soft vote for removing `cni-*-dir` configs until someone needs them.  My pitch is that, until there's a need, we can be opinionated and keep the charm API simple.  My bet is that nobody will need these any time soon.  

If we do keep these, we'll have to figure out what happens when someone specifies `platform=microk8s` and `cni-bin-dir=something`.  I prefer not having to worry about that atm.

Reply:
Agreed, let's defer it until we have a need for it